### PR TITLE
Topic/adapt to gitlab ci

### PR DESCRIPTION
**Pull request by gergondet**
* Provide a gitlab-ci.yml example file
* Implement an option to disable sudo even on travis builds (CI_REQUIRE_SUDO not tested on travis)
* Do not rely on the scripts being in .travis

### CODE REVIEW

**Comment by bchretien** [comment]
:+1: 

With GitLab CI, apt packages are supposed to be installed in those scripts as well? Now that Travis offers a docker-based build, apt packages can be provided in `.travis.yml`, removing the need for sudo in most parts of the script.

**Comment by gergondet** [comment]
For now, apt-dependencies are given in .gitlab-ci.yml in the same way (i.e. by defining the environment variable APT_DEPENDENCIES)  and with the same effect, i.e. trigger apt-get install calls in before_script.

However, since (in our use-case) the gitlab-ci builder is root in its docker container, we do not require the use of sudo. I am unsure about the situation with travis and container-based builds but it may be worth a try although this does not seem to be the proper way to add apt-dependencies in that case.

**Comment by bchretien** [comment]
For travis, we can use this rather than the current `APT_DEPENDENCIES` + `sudo`: http://docs.travis-ci.com/user/apt/

However, sources need to be approved (see their [whitelist](https://github.com/travis-ci/apt-source-whitelist)). This may be a problem for ROS (cf. travis-ci/apt-source-whitelist#171).

For `sudo` requirements, if that only concerns APT packages + `ccache`, we can probably avoid it. We'd need to clean the scripts though, and have different CI-agnostic functions e.g. `install_dependencies()`, `process_coverage()`, `process_doc()`.

File: common.sh [hunk 1]
```diff
@@ -57,4 +103,43 @@ mkdir -p "$build_dir"
...
+
+case "$BASH_VERSION" in
+    '')
+        trap on_exit EXIT
+        ;;
+    *)
+        set -o errtrace
+        trap on_error ERR
+        ;;
+esac
```
Comment (bchretien) [inline common.sh]: Any particular reason for moving this out of `verbose_errors.sh`?

Comment (gergondet) [inline common.sh]: No particular reason (except for a bug earlier in the patch cycle). I've restored this part to the previous version.

**Pull request closed without merging**